### PR TITLE
[feat]: [PLG-3055]: added support for multiple add-on licenses on top of base license

### PR DESCRIPTION
**Pull request by kapil-harness**
## Describe your changes

Testing Scenarios for license add-ons:

1. CSE License CREATE API:
	a. Create an add-on license(isAddOn=TRUE) w/o base license-> should create base license with isAddOn=FALSE
	b. Create multiple add-on licenses-> should allow
	c. Create a base license(isAddOn=FALSE) with an existing base license-> throw exception
	d. Create an add-on license(isAddOn=TRUE) with a base license(isAddOn=FALSE) with greater expiry-> throw exception
	e. Create an add-on license(isAddOn=TRUE) for FREE/TEAM license-> throw exception
2. CSE License UPDATE API:
	a. Update an existing add-on license(isAddOn=TRUE) to base license(isAddOn=FALSE)-> throw exception
	b. Update an existing base license(isAddOn=FALSE) to add-on license(isAddOn=TRUE)-> throw exception
	c. Update an add-on license(isAddOn=TRUE) with a base license(isAddOn=FALSE) with greater expiry-> throw exception
3. CSE License DELETE API:
	a. DELETE an exisitng add-on license-> deletes the add-on license
	b. DELETE an exisitng base license-> deletes the base license if no add-on licenses are present
	c. DELETE an existing base license-> throw exception if atleast 1 add-on license is present
4. CSE License GET API:
	a. successfully return all the licenses(base licenses + add-on licenses)
5. FREE License CREATE API:
	a. successfully create FREE license with isAddOn=FALSE
6. TRIAL License CREATE API:
	a. successfully create TRIAL license with isAddOn=FALSE
7. extend TRIAL License API:
	a. successfully extend TRIAL license with isAddOn=FALSE

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56843)

### CODE REVIEW

**Comment by kapil-harness** [comment]
trigger uts

**Comment by kapil-harness** [comment]
trigger uts

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56843)* status: 7 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/helpers/ModuleLicenseHelper.java` line 241 at r2](https://reviewable.io/reviews/harness/harness-core/56843#-Nlj9UJM6-jG1U_wzDUO:-Nlj9UJM6-jG1U_wzDUP:bugao8w) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/helpers/ModuleLicenseHelper.java#L241)):*
> ```Java
>     }
> 
>     if (update.isAddOn() != current.isAddOn()) {
> ```

can we put this as the first statement in this method as this has a validation associated with it, so it will help to end the flow as soon as possible.

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56843)* status: 7 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 220 at r2](https://reviewable.io/reviews/harness/harness-core/56843#-Nlj9mMS5GbkVBi8EpmP:-Nlj9mMS5GbkVBi8EpmQ:b-cge9qu) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L220)):*
> ```Java
> 
>     ModuleLicense moduleLicense = existingEntityOptional.get();
>     if (moduleLicenseHelper.isDeveloperLicensingFeatureEnabled(existingEntityOptional.get().getAccountIdentifier())) {
> ```

we should use "moduleLicense" instead of "existingEntityOptional.get()"

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56843)* status: 7 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 828 at r2](https://reviewable.io/reviews/harness/harness-core/56843#-NljAAZN4pH4UD9NfiP7:-NljAAZN4pH4UD9NfiP8:beis49y) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L828)):*
> ```Java
>   private void checkForFreeAndTeamModuleLicense(ModuleLicense moduleLicense) {
>     boolean isFreeOrTeamLicense =
>         moduleLicense.getEdition().equals(Edition.FREE) || moduleLicense.getEdition().equals(Edition.TEAM);
> ```

we should do reverse comparisons : Edition.FREE.equals(xyz) or infact Edition.FREE == xyz also works.

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56843)* status: 7 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 845 at r2](https://reviewable.io/reviews/harness/harness-core/56843#-NljAWGj7aGIzYQotRHa:-NljAWGj7aGIzYQotRHb:bevp4xv) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L845)):*
> ```Java
>       } else {
>         throw new InvalidRequestException(String.format(
>             "add-on ModuleLicense with accountIdentifier [%s] and moduleType [%s] and id [%s] does not have a corresponding base ModuleLicense",
> ```

If this is going to be shown to user, then its pretty ambiguous and vague for user understanding.

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56843)* status: 7 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 853 at r2](https://reviewable.io/reviews/harness/harness-core/56843#-NljAgKM1FeDSU-Ql0qf:-NljAgKM1FeDSU-Ql0qg:b-qmbr5g) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L853)):*
> ```Java
>   private void validateAddOnLicenseExpiry(ModuleLicense baseModuleLicense, ModuleLicense moduleLicense) {
>     if (moduleLicense.getExpiryTime() > baseModuleLicense.getExpiryTime()) {
>       throw new InvalidRequestException("add-on ModuleLicense expiry cannot be more than the base ModuleLicense");
> ```

I'm assuming all these are user error messages. So, can we make it more user friendly. eg: We shouldnt use class names here like "ModuleLicense" etc.

File: 930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java [hunk 1]
```diff
@@ -232,14 +238,17 @@ public ModuleLicense getCurrentLicense(String accountId, ModuleType moduleType)
...
-    List<ModuleLicense> existingLicenses = moduleLicenseRepository.findByAccountIdentifierAndModuleType(
-        moduleLicense.getAccountIdentifier(), moduleLicense.getModuleType());
-    if (existingLicenses.size() != 0) {
-      throw new InvalidRequestException(
-          String.format("ModuleLicense with accountIdentifier [%s] and moduleType [%s] already exists",
-              moduleLicense.getAccountIdentifier(), moduleLicense.getModuleType()));
+    if (!moduleLicenseHelper.isDeveloperLicensingFeatureEnabled(moduleLicense.getAccountIdentifier())) {
+      List<ModuleLicense> existingLicenses = moduleLicenseRepository.findByAccountIdentifierAndModuleType(
+          moduleLicense.getAccountIdentifier(), moduleLicense.getModuleType());
+      if (existingLicenses.size() != 0) {
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java:244]: can use `!existingLicenses.isEmpty()` instead

File: 930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java [hunk 2]
```diff
@@ -787,4 +799,71 @@ private void ngAuditModuleLicenseDeleteEvent(ModuleLicenseDTO oldModuleLicenseDT
       log.error("Audit trails for ModuleLicense delete event failed with exception: ", ex);
     }
   }
+
+  private void validateExistingModuleLicense(ModuleLicense moduleLicense) {
+    List<ModuleLicense> existingLicenses = moduleLicenseRepository.findByAccountIdentifierAndModuleType(
+        moduleLicense.getAccountIdentifier(), moduleLicense.getModuleType());
+
+    if (!moduleLicense.isAddOn()) {
+      if (existingLicenses.size() != 0) {
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java:808]: same here

File: 930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java [hunk 3]
```diff
@@ -787,4 +799,71 @@ private void ngAuditModuleLicenseDeleteEvent(ModuleLicenseDTO oldModuleLicenseDT
...
+        throw new InvalidRequestException(
+            String.format("ModuleLicense with accountIdentifier [%s] and moduleType [%s] already exists",
+                moduleLicense.getAccountIdentifier(), moduleLicense.getModuleType()));
+      }
+    } else {
+      checkForFreeAndTeamModuleLicense(moduleLicense);
+
+      Optional<ModuleLicense> baseModuleLicense =
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java:816]: rename it to `maybeModuleLicense` to signify that it's an optional value.

File: 930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java [hunk 4]
```diff
@@ -787,4 +799,71 @@ private void ngAuditModuleLicenseDeleteEvent(ModuleLicenseDTO oldModuleLicenseDT
...
+    if (isFreeOrTeamLicense) {
+      throw new InvalidRequestException("FREE/TEAM ModuleLicense cannot have add-on ModuleLicense");
+    }
+  }
+
+  private void validateUpdateModuleLicense(ModuleLicense moduleLicense) {
+    if (moduleLicense.isAddOn()) {
+      List<ModuleLicense> existingLicenses = moduleLicenseRepository.findByAccountIdentifierAndModuleType(
+          moduleLicense.getAccountIdentifier(), moduleLicense.getModuleType());
+      Optional<ModuleLicense> baseModuleLicense =
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java:838]: same here 
rename it to `maybeModuleLicense` to signify that it's an optional value.

File: 930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java [hunk 5]
```diff
@@ -787,4 +799,71 @@ private void ngAuditModuleLicenseDeleteEvent(ModuleLicenseDTO oldModuleLicenseDT
...
+    if (moduleLicense.getExpiryTime() > baseModuleLicense.getExpiryTime()) {
+      throw new InvalidRequestException("add-on ModuleLicense expiry cannot be more than the base ModuleLicense");
+    }
+  }
+
+  private void validateDeleteModuleLicense(ModuleLicense moduleLicense) {
+    if (!moduleLicense.isAddOn()) {
+      List<ModuleLicense> existingLicenses = moduleLicenseRepository.findByAccountIdentifierAndModuleType(
+          moduleLicense.getAccountIdentifier(), moduleLicense.getModuleType());
+      Optional<ModuleLicense> addOnModuleLicense = existingLicenses.stream().filter(ModuleLicense::isAddOn).findFirst();
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java:861]: rename `addOnModuleLicense` to `maybeAddOnModuleLicense` to signify that it's an optional value.

**Review by kapil-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56843)* status: 7 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/helpers/ModuleLicenseHelper.java` line 241 at r2](https://reviewable.io/reviews/harness/harness-core/56843#-Nlj9UJM6-jG1U_wzDUO:-NljBizADmDcnwlX8JwA:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/helpers/ModuleLicenseHelper.java#L241)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

can we put this as the first statement in this method as this has a validation associated with it, so it will help to end the flow as soon as possible.
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 220 at r2](https://reviewable.io/reviews/harness/harness-core/56843#-Nlj9mMS5GbkVBi8EpmP:-NljDeDU0eq_OzDIuEM7:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L220)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

we should use "moduleLicense" instead of "existingEntityOptional.get()"
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 828 at r2](https://reviewable.io/reviews/harness/harness-core/56843#-NljAAZN4pH4UD9NfiP7:-NljCnUn4otKnP9MRxiK:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L828)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

we should do reverse comparisons : Edition.FREE.equals(xyz) or infact Edition.FREE == xyz also works.
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 853 at r2](https://reviewable.io/reviews/harness/harness-core/56843#-NljAgKM1FeDSU-Ql0qf:-NljEOd39cb_IoKZ6anb:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L853)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

I'm assuming all these are user error messages. So, can we make it more user friendly. eg: We shouldnt use class names here like "ModuleLicense" etc.
</blockquote></details>

Done.

**Review by neelam-harness — CHANGES_REQUESTED** [review]

**Review by kapil-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56843)* status: 7 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 845 at r2](https://reviewable.io/reviews/harness/harness-core/56843#-NljAWGj7aGIzYQotRHa:-NljGiKmBb2taKk56bNQ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L845)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

If this is going to be shown to user, then its pretty ambiguous and vague for user understanding.
</blockquote></details>

Done.

**Review by kapil-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56843)* status: 7 file(s) to review. 2 change requests. 10. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 244 at r2](https://reviewable.io/reviews/harness/harness-core/56843#gh-1428456276:-NljHQy58epUecEnBbyx:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L244)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

can use `!existingLicenses.isEmpty()` instead
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 808 at r2](https://reviewable.io/reviews/harness/harness-core/56843#gh-1428456819:-NljHRiXCCifS0WJmCSt:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L808)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

same here
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 816 at r2](https://reviewable.io/reviews/harness/harness-core/56843#gh-1428457422:-NljHt2x1rar8zOCpwJt:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L816)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

rename it to `maybeModuleLicense` to signify that it's an optional value.
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 838 at r2](https://reviewable.io/reviews/harness/harness-core/56843#gh-1428458109:-NljHubW7PKi8-owf2u5:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L838)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

same here 
rename it to `maybeModuleLicense` to signify that it's an optional value.
</blockquote></details>

Done.
___
*[`930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java` line 861 at r2](https://reviewable.io/reviews/harness/harness-core/56843#gh-1428458944:-NljHw3M4Y7dhYr9wKmg:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/16135b10be2def8019aa754cf1447f28c968b425/930-ng-license-manager/src/main/java/io/harness/licensing/services/DefaultLicenseServiceImpl.java#L861)):*
<details><summary><i>Previously, neelam-harness (Neelam) wrote…</i></summary><blockquote>

rename `addOnModuleLicense` to `maybeAddOnModuleLicense` to signify that it's an optional value.
</blockquote></details>

Done.

**Review by neelam-harness — APPROVED** [review]
LGTM

**Comment by kapil-harness** [comment]
trigger uts

File: 930-ng-license-manager/src/test/java/io/harness/licensing/services/DefaultLicenseServiceImplTest.java [hunk 6]
```diff
@@ -134,6 +136,7 @@ public void setUp() {
     startTrialRequestDTO = StartTrialDTO.builder().moduleType(DEFAULT_MODULE_TYPE).edition(Edition.ENTERPRISE).build();
     when(licenseObjectConverter.toDTO(DEFAULT_CI_MODULE_LICENSE)).thenReturn(DEFAULT_CI_MODULE_LICENSE_DTO);
     when(licenseObjectConverter.toEntity(DEFAULT_CI_MODULE_LICENSE_DTO)).thenReturn(DEFAULT_CI_MODULE_LICENSE);
+    when(moduleLicenseHelper.isDeveloperLicensingFeatureEnabled(any())).thenReturn(false);
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/test/java/io/harness/licensing/services/DefaultLicenseServiceImplTest.java:139]: can we add tests for the True case as well ?

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 4 of 6 files at r1, 1 of 1 files at r2, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/56843)* status: 7 file(s) to review. 5. unresolved discussions.

**Comment by kapil-harness** [comment]
trigger review

**Pull request merged**